### ISSUE TITLE
Overflowing labels  + not working keybinds

### ISSUE THREAD

**Issue by georgi-chinov** [Sellorio/mega-macro#166]
Few things I spotted after the curse version being released:

![image](https://user-images.githubusercontent.com/9299216/198581966-ccd738c8-502c-4204-9045-49d0f5fee853.png)
Macro overflow outside the box

Secondly some macros don't work on pressing the keybind. They do work on clicking the macro on the bar tho. Not sure if not a blizzard bug

**Comment by aurelion314** [comment]
> Secondly some macros don't work on pressing the keybind. They do work on clicking the macro on the bar tho. Not sure if not a blizzard bug

I believe this is due to a combination of blizzard's new hold and release mechanic, combined with having other spells or items in the macro. I had this on a couple spells, but when I removed my on-use trinket from the macro, it worked again.

I also noticed that you can still use the on-use in the macro if you use '@cursor' in the spell modifier (so it isn't waiting for the click), and switch the option in the settings from hold and release, to press and click.  This does mean you can't hold it though, so may not work for everything. 

I heard this happens even with Blizzards default macro's (haven't tested myself), so I don't think there is much we can do about it. If the issue is something else, can you show the macro that doesn't work?

**Comment by socalminstrel** [comment]
For me, even clicking on the macro on the action bar does nothing. Non-macro'd abilities work fine, clicked or with keybind. I still get the "button being pushed" border on the action bar button when I use a keybind for a macro, but the macro doesn't actually fire. This is true for all macros.

**Comment by aurelion314** [comment]
> For me, even clicking on the macro on the action bar does nothing.

Have you altered the setting "ActionButtonUseKeyDown"? If you change that in game, then you need to reload the UI for macros to work. The latest version of the addon auto detect the value and works around it, but cannot change it on the fly, so a reload is required.

You can manually change it with `/console ActionButtonUseKeyDown 1`

**Comment by socalminstrel** [comment]
Thanks for the reply. I did reload it before, not for that reason which I wasn't aware of, just in case it needed a reload after initial set up. I've also logged out and back in.

**Comment by aurelion314** [comment]
Hmm, can you try changing the value with the command I showed? (try setting to 0, reload, try 1, reload)

Does right clicking the macro still work?

**Comment by jlemerle** [comment]
`/console ActionButtonUseKeyDown 1`

This seems to be break the addon. You are no longer able to press any of the macro's you made within Mega Macro, even after `/reload`

Setting `/console ActionButtonUseKeyDown 0` allows the macros to be used again.

Tested this without Mega Macro and with `/console ActionButtonUseKeyDown 1` the macro's work as normal.

**Comment by socalminstrel** [comment]
> Hmm, can you try changing the value with the command I showed? (try setting to 0, reload, try 1, reload)
> 
> Does right clicking the macro still work?

The macros don't work normally with either setting and right-clicking the macros don't work either. Oddly, though, the macros will fire once after a reload and then never again--until you reload again.

**Comment by aurelion314** [comment]
Huh that is strange. Are there any Lua errors showing up (`/console ScriptErrors 1`)? If the addon is loading properly I have no idea why it would do that. Maybe a conflict with another addon?

**Comment by socalminstrel** [comment]
Yup, conflict with another addon. I feel silly not testing it with all other addons off. Thanks for the help. I guess I'll figure what addon it is (maybe Bartender or something) and see what I can do about it.

**Comment by PunkROC01** [comment]
It looks like it is ElvUI that breaks my macros when Key Down is active. ElvUI recently made some changes because Key Down was interfering with Top Auras not being right click cancellable. So they removed the Key Down option

When I disable ElvUI, but use Key Down, the macros work fine. And since ElvUI doesn't support downgrading (can't even find old versions anywhere) I am stuck since my whole UI is built around ElvUI

**Comment by georgi-chinov** [comment]
Don't think it's elvui issue. I disabled all addons but mega macro and bug sack + grabber.

I think i figured out the problem, macros with [nomod] etc  don't work but I think that's due to new blizzard macro API

**Comment by PunkROC01** [comment]
Ah, interesting. None of my macros use "nomod" or "mod:alt/shift/ctrl". But it was ElvUI causing my macros not to work when Key Down was active. However, I fixed that by editing line 123 back to 

"/click [btn:1] "..ClickyFrameName..macroId.." LeftButton\n"..

So now everything works fine for me

**Comment by georgi-chinov** [comment]
In which file you made that change ?

**Comment by jlemerle** [comment]
> Hi! Don't think it's elvui issue. I disabled all addons but mega macro and bug sack + grabber.
> 
> I think i figured out the problem, macros with [nomod] etc don't work but I think that's due to new blizzard macro API ![image](https://user-images.githubusercontent.com/9299216/198826483-6f8a487d-6495-40d4-a9b2-ee02b651f038.png) this works ![image](https://user-images.githubusercontent.com/9299216/198826488-bf8960d5-7a56-4935-a55f-f64c8d7a8249.png) this doesn't
> 
> edit: after bit more testing, pressing alt and clicking second macro works, just the keypress doesn't

Exactly the same issue as me, good catch on the [mod:xxxx] and [nomod], every other macro works fine, without it.

Sadly, I am so dependent on the modifier usage :(

**Comment by socalminstrel** [comment]
> Exactly the same issue as me, good catch on the [mod:xxxx] and [nomod], every other macro works fine, without it.
> 
> Sadly, I am so dependent on the modifier usage :(

Something else is at play here too, because all my modifier macros work fine, including ones with the [nomod] conditional.

**Comment by georgi-chinov** [comment]
Ok ok finally figured it out!

Bind for macro is ctr + 1 so it counts the control as modifier.  It wasn't before so maybe it's some configuration in the addon that needs to be touched..

**Comment by jlemerle** [comment]
> > Exactly the same issue as me, good catch on the [mod:xxxx] and [nomod], every other macro works fine, without it.
> > Sadly, I am so dependent on the modifier usage :(
> 
> Something else is at play here too, because all my modifier macros work fine, including ones with the [nomod] conditional.

ElvUI + MegaMacro = **Working** with modifier macros
ElvUI + MegaMacro + `/console ActionButtonUseKeyDown 1` = Not working with modifier macros
MegaMacro + `/console ActionButtonUseKeyDown 1` = **Working** with modifier macros

**Comment by georgi-chinov** [comment]
what I discovered is not related to elv ui or /console ActionButtonUseKeyDown 1

The addon thinks ctr is modifier when ctr+1 is the keybind. SInce it was working before like this there has to be something in the addon to recognize wieth a modifier or keybind is pressed

**Comment by Fizgig1973** [comment]
Stil not working for me, at least with keybinds. Pressing it once works, after that, or after zoning they just turn into question marks. Not using any modifiers, or mods with keybinds.

**Comment by georgi-chinov** [comment]
A workaround I came up with: 
![image](https://user-images.githubusercontent.com/9299216/198896430-b81e1755-aea5-4ee9-908a-0a70092230ac.png)
